compiler/rustc_baked_icu_data/src/data/macros.rs

```diff
@@ -18,29 +18,29 @@ macro_rules! __make_provider {
         impl $name {
             #[doc(hidden)]
             #[allow(dead_code)]
-            pub const MUST_USE_MAKE_PROVIDER_MACRO: () = ();
```

Comment:
1st line contains `// @generated`, so probably this file shouldn't be changed?

Reply:
Indeed, `rustc_baked_icu_data` is mostly auto-generated (except for `lib.rs`)[^1]. We should not be modifying this crate (at least not directly). I think we should just allow the lint for this crate until the generator is updated.

[^1]: https://github.com/rust-lang/rust/blob/413c9bf364f84939e05bf9d5abdccb0d6abf2fc3/compiler/rustc_baked_icu_data/src/lib.rs#L1-L4

Reply:
Fixed.